File: LeetCode/keyboard_row.cpp
class Solution {
public:
    vector<string> findWords(vector<string>& words) {
        bool valid = true;
        int i;
        set<char> row1;
        set<char> row2;
        set<char> row3;
        char  r1[]= {'q','w','e','r','t','y','u','i','o','p','Q','W','E','R','T','Y','U','I','O','P'};
        row1.insert (r1,r1+20);
        
        char r2[] = {'a','s','d','f','g','h','j','k','l','A','S','D','F','G','H','J','K','L'};
        row2.insert(r2,r2+18);
        
        char r3[] = {'z','x','c','v','b','n','m','Z','X','C','V','B','N','M'};
        row3.insert(r3,r3+14);
        
        vector<string> result;
        
        for(int iter = 0, size = words.size(); iter< size;iter++){
            valid = true;
            for(i = 0; i < words[iter].length(); i++){
                if(row1.find(words[iter][i]) != row1.end() ){
                    continue;
                }
                else{
                    break;
                }
            }
            if(i == words[iter].length()){
                result.push_back(words[iter]);
            }
            //checking row 2
            for(i = 0; i < words[iter].length(); i++){
                if(row2.find(words[iter][i]) != row2.end() ){
                    continue;
                }
                else{
                    break;
                }
            }
            if(i == words[iter].length()){
                result.push_back(words[iter]);
            }
            //checking row 3
            for(i = 0; i < words[iter].length(); i++){
                if(row3.find(words[iter][i]) != row3.end() ){
                    continue;
                }
                else{
                    break;
                }
            }
            if(i == words[iter].length()){
                result.push_back(words[iter]);
            }
            
        }
        return result;
    }
};

File: LeetCode/keyboard_row.py
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        firstRow = "qwertyuiop"
        secondRow = "asdfghjkl"
        thirdRow = "zxcvbnm"

        d = {}
        for c in firstRow:
            d.update({c: 1})
        for c in secondRow:
            d.update({c: 2})
        for c in thirdRow:
            d.update({c: 3})

        result = []

        for wordValue in words:
            word = wordValue.lower()
            homeRow = d.get(word[0])
            valid = True
            for c in word:
                if homeRow != d.get(c):
                    print word + " is not in same row"
                    valid = False
                    break
            if valid:
                result.append(wordValue)

        return result
